File: WoB_Client/Assets/Network/Response/ResponseTopList.cs
/**
 * From: Lobby team
 * Author: Jacob
 * Description of Class: response structure for receiving toplist from the server
***/


using UnityEngine;

using System;
using System.Collections.Generic;

public class ResponseTopListEventArgs : ExtendedEventArgs {
	public string name1 { get; set; }
	public int score1 { get; set; }
	public string name2 { get; set; }
	public int score2 { get; set; }
	public string name3 { get; set; }
	public int score3 { get; set; }

	public ResponseTopListEventArgs() {
		event_id = Constants.SMSG_TOPLIST;
	}
}

public class ResponseTopList : NetworkResponse {
	
	public string name1;
	public int score1;
	public string name2;
	public int score2;
	public string name3;
	public int score3;
	
	public ResponseTopList() {
	}
	
	public override void parse() {
		name1 = DataReader.ReadString(dataStream);
		score1 = DataReader.ReadInt(dataStream);
		name2 = DataReader.ReadString(dataStream);
		score2 = DataReader.ReadInt(dataStream);
		name3 = DataReader.ReadString(dataStream);
		score3 = DataReader.ReadInt(dataStream);
	}
	
	public override ExtendedEventArgs process() {
		ResponseTopListEventArgs args = new ResponseTopListEventArgs();
		
		args.name1 = name1;
		args.score1 = score1;
		args.name2 = name2;
		args.score2 = score2;
		args.name3 = name3;
		args.score3 = score3;
		
		return args;
	}
}

File: WoB_Server/src/networking/response/ResponseTopList.java
/**
 * From: Lobby team
 * Author: Jacob
 * Description of Class: response structure for receiving toplist from the server
***/

package networking.response;

import metadata.Constants;
import utility.GamePacket;

public class ResponseTopList extends GameResponse {

    private String name1;
    private int score1;
    private String name2;
    private int score2;
    private String name3;
    private int score3;

    public ResponseTopList() {
        responseCode = Constants.SMSG_TOPLIST;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        
        //TODO server team: construct packet w/ toplist info

        return packet.getBytes();
    }
}
